main.tf

```diff
@@ -286,7 +287,7 @@ resource "aws_alb_listener" "test_listener" {
   load_balancer_arn = aws_alb.alb.arn
   port              = var.codedeploy_test_listener_port
   protocol          = "HTTPS"
-  certificate_arn   = var.https_certificate_arn
+  certificate_arn   = local.create_new_https_cert ? aws_acm_certificate.new_cert[0].arn : var.https_certificate_arn # if cert is not provided use created one, else use existing cert
```

Comment:
This should use `aws_acm_certificate_validation.new_cert.certificate_arn` so that the Terraform doesn't try to create the listener before the cert is ready to be used.

Reply:
done